### PR TITLE
[Feature]Run kuberay in a single namespace

### PR DESCRIPTION
**Pull request by wilsonwang371**
## Why are these changes needed?

Run kuberay in a single namespace.

Instead of  setup kuberay using 
```bash
kubectl apply -k "github.com/ray-project/kuberay/manifests/cluster-scope-resources"
kubectl apply -k "github.com/ray-project/kuberay/manifests/base"
```
now we use
```bash
kubectl apply -k "github.com/ray-project/kuberay/manifests/cluster-scope-resources"
kubectl apply -k "github.com/ray-project/kuberay/manifests/singe-namespace"
```

## Related issue number

#227

## Checks

- [ ] I've made sure the tests are passing. 
- Testing Strategy
   - [ ] Unit tests
   - [X] Manual tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by wilsonwang371** [comment]
cc @Jeffwan @scarlet25151

**Comment by Jeffwan** [comment]
image

Can we revert the rename changes? The major reason is the current folder structure is from kubebuilder output. Sometimes, when we upgrade the kubebuilder etc, we still like the structure to be working and that's why we want them remain the same.

**Comment by Jeffwan** [comment]
For the changes here, I feel we can override the setting at the root kustomization level? 
image

**Comment by Jeffwan** [comment]
Let's have a discussion tomorrow on this.

**Comment by wilsonwang371** [comment]
now we use kustomize to patch clusterrole clusterrolebinding with a very minimal code change.

cc @Jeffwan

File: manifests/overlays/single-namespace/kustomization.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# This overlay patches in KubeRay operator configuration
+# necessary for KubeRay single namespace support.
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+
+bases:
+- ../../base
+patches:
+- path: patch-cr.json
```
Comment (Jeffwan) [inline manifests/overlays/single-namespace/kustomization.yaml:9]: What about the name? Will we have `cluster-role` created in namespace scope?

Comment (wilsonwang371) [reply manifests/overlays/single-namespace/kustomization.yaml:9]: this will convert all cluster-role to role with namespace.

**Comment by wilsonwang371** [comment]
> > kubectl apply -k "github.com/ray-project/kuberay/manifests/cluster-scope-resources"
> > kubectl apply -k "github.com/ray-project/kuberay/manifests/single-namespace"
> 
> I guess an admin may be the one to run the first step. A user with restricted permissions would run the second step.

yes, I should make this more clear. I updated the README document

**Pull request merged**